### PR TITLE
Support Tracing

### PR DESCRIPTION
**Pull request by Yannic92**
Fixes: #1135

### CODE REVIEW

File: internal/utils/metrics/src/main/java/org/eclipse/ditto/internal/utils/metrics/instruments/timer/OnStopHandler.java [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
- * SPDX-License-Identifier: EPL-2.0
- */
-package org.eclipse.ditto.internal.utils.metrics.instruments.timer;
-
-import java.util.function.Consumer;
-
-/**
- * Contains {@code stoppedTimerConsumer} to be invoked when a Timer stops.
- */
-final class OnStopHandler {
```
Comment (Yannic92) [inline internal/utils/metrics/src/main/java/org/eclipse/ditto/internal/utils/metrics/instruments/timer/OnStopHandler.java:20]: what was the reason to delete this class? I think there's no need to have it as class but mabye as functional interface with `handleStoppedTimer` beeing the method?

Comment (Yannic92) [reply internal/utils/metrics/src/main/java/org/eclipse/ditto/internal/utils/metrics/instruments/timer/OnStopHandler.java:20]: My personal opinion is that I don't like increasing abstraction.

**Review by Yannic92 — COMMENTED** [review]
Just some open questions to maybe rethink some decisions.

File: bom/pom.xml [hunk 2]
```diff
@@ -65,9 +65,10 @@
         <janino.version>2.7.8</janino.version>
 
         <!-- ### Metrics and Tracing -->
-        <kamon-core.version>2.1.4</kamon-core.version>
-        <kamon-system-metrics.version>2.1.4</kamon-system-metrics.version>
-        <kamon-prometheus.version>2.1.4</kamon-prometheus.version>
+        <kamon-core.version>2.2.2</kamon-core.version>
+        <kamon-system-metrics.version>2.2.2</kamon-system-metrics.version>
+        <kamon-prometheus.version>2.2.2</kamon-prometheus.version>
+        <kamon-opentelemetry.version>2.2.2</kamon-opentelemetry.version>
```
[the remark this answers is not in the document]
Comment (dguggemos) [reply bom/pom.xml:71]: Yes, makes sense.

File: internal/utils/metrics/src/main/java/org/eclipse/ditto/internal/utils/metrics/instruments/timer/OnStopHandler.java (hunk 1 above)
Comment (dguggemos) [reply internal/utils/metrics/src/main/java/org/eclipse/ditto/internal/utils/metrics/instruments/timer/OnStopHandler.java:20]: > I think there's no need to have it as class but mabye as functional interface with handleStoppedTimer beeing the method

Sounds like a good compromise 😃

File: base/service/src/main/java/org/eclipse/ditto/base/service/DittoService.java [hunk 3]
```diff
@@ -247,15 +246,18 @@ private void startKamon() {
...
+        if (metricsConfig.isSystemMetricsEnabled() || tracingConfig.isTracingEnabled()) {
             // start system metrics collection
             Kamon.init();
         }
         if (metricsConfig.isPrometheusEnabled()) {
             // start prometheus reporter
             startPrometheusReporter();
         }
+
+        DittoTracing.initialize(tracingConfig);
```
[the remark this answers is not in the document]
Comment (dguggemos) [reply base/service/src/main/java/org/eclipse/ditto/base/service/DittoService.java:260]: The check is done in the initialize method. Would you rather do it outside of it?

[the remark this answers is not in the document]
Comment (dguggemos) [reply base/service/src/main/java/org/eclipse/ditto/base/service/DittoService.java:260]: We agreed that the current solution is fine.

**Review by thjaeckle — APPROVED** [review]

**Pull request merged**